cs-google/src/main/scala/io/cloudslang/content/google/actions/compute/compute_engine/instances/InstancesGetSerialPortOutput.scala

```diff
@@ -0,0 +1,112 @@
+    * @param startIndexInp    Optional - The byte position from which to return the output. Use this to page through output
+    *                         when the output is too large to return in a single request. For the initial request, leave
+    *                         this field unspecified. For subsequent calls, this field should be set to the next value
+    *                         returned in the previous call.
+    *                         Default: 0
+    * @param proxyHost        Optional - Proxy server used to connect to Google Cloud API. If empty no proxy will
+    *                         be used.
+    * @param proxyPortInp     Optional - Proxy server port used to access the provider services.
+    *                         Default: "8080"
+    * @param proxyUsername    Optional - Proxy server user name.
+    * @param proxyPasswordInp Optional - Proxy server password associated with the <proxyUsername> input value.
+    * @return the Serial Port Output as returnResult and the nextIndex with the last byte position read
+    */
+  @Action(name = "Get Serial Port Output",
+    outputs = Array(
+      new Output(RETURN_CODE),
+      new Output(RETURN_RESULT),
+      new Output(EXCEPTION),
+      new Output(NEXT_INDEX)
+    ),
+    responses = Array(
+      new Response(text = ResponseNames.SUCCESS, field = RETURN_CODE, value = ReturnCodes.SUCCESS, matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED),
+      new Response(text = ResponseNames.FAILURE, field = RETURN_CODE, value = ReturnCodes.FAILURE, matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR, isOnFail = true)
+    )
+  )
+  def execute(@Param(value = PROJECT_ID, required = true) projectId: String,
+              @Param(value = ZONE, required = true) zone: String,
+              @Param(value = ACCESS_TOKEN, required = true, encrypted = true) accessToken: String,
+
+              @Param(value = INSTANCE_NAME, required = true) instanceName: String,
+              @Param(value = CONSOLE_PORT) consolePortInp: String,
+              @Param(value = START_INDEX) startIndexInp: String,
+
+              @Param(value = PROXY_HOST) proxyHost: String,
+              @Param(value = PROXY_PORT) proxyPortInp: String,
+              @Param(value = PROXY_USERNAME) proxyUsername: String,
+              @Param(value = PROXY_PASSWORD, encrypted = true) proxyPasswordInp: String): util.Map[String, String] = {
+    val proxyHostOpt = verifyEmpty(proxyHost)
+    val proxyUsernameOpt = verifyEmpty(proxyUsername)
+    val proxyPortStr = defaultIfEmpty(proxyPortInp, DEFAULT_PROXY_PORT)
+    val proxyPassword = defaultIfEmpty(proxyPasswordInp, EMPTY)
+
+    val consolePortStr = defaultIfEmpty(consolePortInp, DEFAULT_CONSOLE_PORT)
+    val startIndexStr = defaultIfEmpty(startIndexInp, DEFAULT_START_INDEX)
+
+    val validationStream = validateProxyPort(proxyPortStr) ++
+      validateNonNegativeInteger(consolePortStr, CONSOLE_PORT) ++
```

Comment:
shouldn't we use a more strict validation here ?

Reply:
maybe they want to add more ports ... we don't know the evolution of the sdk